_transcripts/031-rust-1.46-1.47.md

```diff
@@ -0,0 +1,894 @@
+Guard, there's a little diagram, and basically what it does is, it injects a
+little bit of code into your binary that does some runtime checks to make sure
+that if someone, like, attacks your program and manages to overwrite a buffer or
+something, they can't easily take over control of the control flow of your
+program. So this would be something like, imagine that you take a function
+pointer F, and you call F through that function pointer. And then imagine that
+some attacker manages to overwrite F with an address of, like, some evil code
+they want to run instead. What CFG does, the Control Flow Guard, is basically
+inject this little check, just before the call to F, to check that F is still a
+valid target, that it doesn't point into some, like, bad memory or something.
+And if it does, then it just terminates the process. There's a little bit of
+sort of coordination that's required here between the compiler and the Windows
+kernel at runtime that keeps track of, like, what are valid jump targets, like
+what are valid functions to call, and such. So it basically prevents some
+attacks that are bad when they happen. I don't know how effective it is at
+preventing real attacks, like in the wild, but it certainly seems like something
+that that could be a pretty good idea, to just—
+
+__Ben__: I'm sure it's based on things that have been observed in the wild. I
+know that, like, it might not be as useful for Rust in particular, in terms of,
+like, if you're writing `unsafe` code in Rust, I'm sure it is, but it's not
+actually turned on by default in `rustc`, because I think the idea is that
+almost all Rust code actually doesn't need this. Obviously, there are plenty of
+things in Rust, that Rust does turn on by default. And so you have, like,
+typical hardening things, like position independent code and all the other— we
+have stack guards and stack canaries and things that kind of like, even though
+normally you wouldn't encounter these. Sometimes it's just good to have for
+additional security. But I think in this case, it was just decided that the
+performance tradeoff to actual risk was not good enough to turn it on. But
+obviously, you can turn it on if you so elect to, with this new flag to `rustc`.
+
+__Jon__: Yeah, that sounds about right.
+
+There were a bunch of smaller library changes in 1.47 as well. And I think
+before we go to some of the specifics, do you want to talk a little bit about
+the `const` changes here?
+
+__Ben__: Yeah. So as we mentioned before in the previous release, `const` now
+has the capability of doing a bunch of `if`s and `loop`s. And so there are
+plenty of newly-const methods. In this case, although the scope is pretty
+narrow, the biggest group are probably: for all integers, there are various
+operations. There's `checked_add`, `checked_sub`-traction,
+`checked_mul`-tiplication, and by "checked" we mean that in this case, if it
```

Comment:
My translation of "checked subtraction, checked multiplication" may be questionable here? (audio 34:56)

Maybe I should have just dropped the -traction, -tiplication instead?

Reply:
No, I like it!